plugin/src/generators/lib-move/generator.ts

```diff
@@ -0,0 +1,71 @@
+import { Tree } from '@nx/devkit';
+import { LibMoveGeneratorSchema } from './schema';
+import {
+  LibraryType,
+  askQuestion,
+  constants,
+  dynamicImport,
+  filterSource,
+  getLibrariesDetails,
+  selectApplication,
+  selectLibrary,
+  validateLibraryType,
+  getLibDirectoryName
+} from '../../shared/utils';
+import { execSync } from 'child_process';
+import * as path from 'path';
+
+export async function libMoveGenerator(
+  tree: Tree,
+  options: LibMoveGeneratorSchema
+) {
+  const { default: autocomplete } = await dynamicImport<typeof import('inquirer-autocomplete-standalone')>('inquirer-autocomplete-standalone');
+
+  let srcLibraryName: string;
+  let srcLibraryPath: string;
+
+  if (options.srcLibName) {
+    srcLibraryName = options.srcLibName;
+
+    const library = getLibrariesDetails(tree).find((library) => library.name === srcLibraryName);
+
+    if (!library) {
+      throw new Error(`Library ${srcLibraryName} not found`);
+    }
+
+    srcLibraryPath = library.path;
+  } else {
+    const selectedLibrary = await selectLibrary(tree, 'Select the library to move: ');
+
+    srcLibraryName = selectedLibrary.name;
+    srcLibraryPath = selectedLibrary.path;
+  }
+  
+  const libPathSegments = srcLibraryPath.split('/');
+  const defaultDestLibraryName = libPathSegments.pop();
```

Comment:
This code is also in `libRenameGenerator`. Is it possible to make a shared function with it?

Reply:
Done. I did not include `libPathSegments` because in the `lib-rename` it has more manipulations